modules/social_features/social_profile/social_profile.module

```diff
@@ -831,3 +832,19 @@ function social_profile_social_user_account_header_items_alter(array &$menu_item
     ->view($profile, 'small');
 
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ */
+function social_profile_profile_access(EntityInterface $entity, $operation, AccountInterface $account) {
+
+  // Provides access only to viewing user profile info, like referenced data in
+  // featured items, if the current user has no permissions.
+  if (
+    $operation === 'view' &&
```

Comment:
@ribel @volodymyr-sydor-lemberg have we checked any other potential impact?

What happens on the search now for example? We're now checking all routes except a users profile itself. 
Can't find if you've tested / reviewed the unhappy scenario's on this PR.

Reply:
@ronaldtebrake good catch! Yes, we've checked unhappy scenarios for user stream/information/profile pages but on the search/all we can get user items in the result and need to add this route into exceptions. 
 Also, there are pages like:
- all-members
- birthday (social_birthday module)
we can add them too but they are not accessible for anonymous

Reply:
Thanks for the explanation, great to hear! :D 
Maybe in this case it's safer to check if it's part of a landing page or dashboard?
In that case we can always add to it when necessary or expose it as a setting in the future. 
Because our community might have even more routes which they want/need to add into exceptions.

Reply:
I'm afraid that in this `hook_entity_access` we don't have any info about a place where exactly it used
About the list of disabled routes, I think we can implement it as a `hook_disable_view_teaser_profile_routes()` which returns a list of exceptions and can be altered by other modules

Reply:
Sorry, didn't get that completely out of the PR.

What I want to make sure is that in a minor release we are not adding a security risk like this. Not every user has the ability to create custom code. Right now site builders are in no way possible to deny access. Which is my main concern.

So maybe it would be better to create a custom permission or a site setting so people can remove that access? 
What if there is no public content for example? Or if they are not using landing pages / dashboard. 

We don't want them to get an access allowed for this scenario and them not having an option to deny access by a setting or a permission or something you can use the UI for.

Do you see my point? 😊

Reply:
Hmm no using that permission seems ok 🤔  ok check that seems to be a good answer :D thanks for the explanation not sure why i missed that haha

Reply:
Hey @ronaldtebrake, I've improved it a bit and added an additional condition to give access AN to preview teasers profiles  only for landing & dashboard pages, please take a look